lib/pure/json.nim

```diff
@@ -903,29 +903,45 @@ proc parseJson(p: var JsonParser; rawIntegers, rawFloats: bool, depth = 0): Json
   of tkCurlyLe:
     if depth > DepthLimit:
       raiseParseErr(p, "}")
-    result = newJObject()
-    discard getTok(p)
-    while p.tok != tkCurlyRi:
-      if p.tok != tkString:
-        raiseParseErr(p, "string literal as key")
-      var key = p.a
-      discard getTok(p)
-      eat(p, tkColon)
-      var val = parseJson(p, rawIntegers, rawFloats, depth+1)
-      result[key] = val
-      if p.tok != tkComma: break
+    var obj: JsonNode = nil
+    try:
+      obj = newJObject()
       discard getTok(p)
-    eat(p, tkCurlyRi)
+      while p.tok != tkCurlyRi:
+        if p.tok != tkString:
+          raiseParseErr(p, "string literal as key")
+        var key = p.a
+        discard getTok(p)
+        eat(p, tkColon)
+        var val = parseJson(p, rawIntegers, rawFloats, depth+1)
+        obj[key] = val
+        if p.tok != tkComma: break
+        discard getTok(p)
+      eat(p, tkCurlyRi)
+      result = obj
+      obj = nil
+    except JsonParsingError:
+      if obj != nil:
+        obj = nil
```

Comment:
If this is required there is an arc bug. One should not have to play GC to get non leaking code.